src/rpc/handlers/GetAggregatePrice.cpp

```diff
@@ -0,0 +1,313 @@
+#include <optional>
+#include <string>
+#include <variant>
+
+namespace rpc {
+
+GetAggregatePriceHandler::Result
+GetAggregatePriceHandler::process(GetAggregatePriceHandler::Input input, Context const& ctx) const
+{
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence
+    );
+
+    if (auto const status = std::get_if<Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<ripple::LedgerHeader>(lgrInfoOrStatus);
+
+    // sorted descending by lastUpdateTime, ascending by AssetPrice
+    using TimestampPricesBiMap = boost::bimaps::bimap<
+        boost::bimaps::multiset_of<std::uint32_t, std::greater<std::uint32_t>>,
+        boost::bimaps::multiset_of<ripple::STAmount>>;
+
+    TimestampPricesBiMap timestampPricesBiMap;
+
+    for (auto const& oracle : input.oracles) {
+        auto const oracleIndex = ripple::keylet::oracle(oracle.account, oracle.documentId).key;
+
+        auto const oracleObject = sharedPtrBackend_->fetchLedgerObject(oracleIndex, lgrInfo.seq, ctx.yield);
+        if (not oracleObject)
+            continue;
+
+        ripple::STLedgerEntry const oracleSle{
+            ripple::SerialIter{oracleObject->data(), oracleObject->size()}, oracleIndex
+        };
+
+        tracebackOracleObject(ctx.yield, oracleSle, [&](auto const& node) {
+            auto const& series = node.getFieldArray(ripple::sfPriceDataSeries);
+            // Find the token pair entry with the price
+            if (auto const iter = std::find_if(
+                    series.begin(),
+                    series.end(),
+                    [&](ripple::STObject const& o) -> bool {
+                        return o.getFieldCurrency(ripple::sfBaseAsset).getText() == input.baseAsset and
+                            o.getFieldCurrency(ripple::sfQuoteAsset).getText() == input.quoteAsset and
+                            o.isFieldPresent(ripple::sfAssetPrice);
+                    }
+                );
+                iter != series.end()) {
+                auto const price = iter->getFieldU64(ripple::sfAssetPrice);
+                // Asset price is after scale, so we need to get the negative of the scale
+                auto const scale =
+                    iter->isFieldPresent(ripple::sfScale) ? -static_cast<int>(iter->getFieldU8(ripple::sfScale)) : 0;
+
+                timestampPricesBiMap.insert(TimestampPricesBiMap::value_type(
+                    node.getFieldU32(ripple::sfLastUpdateTime), ripple::STAmount{ripple::noIssue(), price, scale}
+                ));
+                return true;
+            }
+            return false;
+        });
+    }
+
+    if (timestampPricesBiMap.empty())
+        return Error{Status{ripple::rpcOBJECT_NOT_FOUND}};
+
+    auto const latestTime = timestampPricesBiMap.left.begin()->first;
+
+    Output out;
```

Comment:
Maybe it would be better to have constructor for `Output` to not forget to fill some field?